phoenix-core/src/main/java/org/apache/phoenix/schema/SaltingUtil.java

```diff
@@ -123,4 +130,29 @@ public static void addRegionStartKeyToScanStartAndStopRows(byte[] startKey, byte
             scan.setStopRow(newStopRow);
         }
     }
+
+    public static void checkTableIsSalted(Admin admin, TableName tableName, int buckets) throws IOException, SQLException {
+        // Heuristics to check if and existing HBase table can be a salted Phoenix table
+        // with the specified number of buckets.
+        // May give false negatives.
+        byte[] maxStartKey = new byte[] {0} ;
+        List<RegionInfo> regionInfos =
+                admin.getRegions(tableName);
+        for (RegionInfo regionInfo : regionInfos) {
+            if (Bytes.compareTo(regionInfo.getStartKey(), maxStartKey) > 0) {
+                maxStartKey = regionInfo.getStartKey();
+            }
+        }
+        byte lastRegionStartSaltKey = maxStartKey[0];
+        if (lastRegionStartSaltKey == (byte) (buckets - 1)) {
```

Comment:
I like the vision of what you are doing here. It's 100% OK to me to try to catch the obvious case(s) that we can and not try to catch all possibilities for messed up region boundaries + salt.

Is it possible to add an "escape hatch" to this method for an experienced user to give a "I know what I'm doing" override? I can't think of a case where your logic is insufficient, but could envision a scenario where something unexpected happens ;)

Reply:
Technically it is possible, but it would be a lot of work as we'd need to start from the grammar to add a force option.

I've handled a few such cases, and in every case it was a user error.
I have given this some thought, and I cannot imagine such a scenario.

If the user has merged his last regions on his table that otherwise matches the salt bucket count, and wants to create the table, it still wouldn't work correctly, and they are better off fixing their splits first, and re-creating the Phoenix table second.

Reply:
> Technically it is possible, but it would be a lot of work as we'd need to start from the grammar to add a force option.

I see that sqlline has some functionality to pass down generic properties, but we don't have that hooked up through sqlline.py

Can defer this until it bites us, I guess.